src/filesystem.cpp

```diff
@@ -66,10 +66,15 @@ bool do_mkdir( std::string const &path, int const mode )
 bool assure_dir_exist( std::string const &path )
 {
     DIR *dir = opendir( path.c_str() );
+    size_t slash_idx = 0;
     if( dir != nullptr ) {
         closedir( dir );
         return true;
     }
+    slash_idx = path.find_last_of( "\\/" );
+    if ( slash_idx != std::string::npos ) {
+        assure_dir_exist( path.substr( 0, path.find_last_of( "\\/" ) ) );
```

Comment:
Isn't this the same value as stored in `slash_idx`? Why not reuse that value here?

Reply:
Yep I seemed to have forgotten to put it in, thanks!